src/client-side-encryption/stateMachine.ts

```diff
@@ -43,90 +53,135 @@ const INSECURE_TLS_OPTIONS = [
 ];
 
 /**
- * @ignore
- * @callback StateMachine~executeCallback
- * @param {Error} [err] If present, indicates that the execute call failed with the given error
- * @param {object} [result] If present, is the result of executing the state machine.
- * @returns {void}
+ * Helper function for logging. Enabled by setting the environment flag MONGODB_CRYPT_DEBUG.
+ * @param msg - Anything you want to be logged.
  */
+export function debug(msg: unknown) {
+  if (process.env.MONGODB_CRYPT_DEBUG) {
+    // eslint-disable-next-line no-console
+    console.error(msg);
+  }
+}
 
-/**
- * @ignore
- * @callback StateMachine~fetchCollectionInfoCallback
- * @param {Error} [err] If present, indicates that fetching the collection info failed with the given error
- * @param {object} [result] If present, is the fetched collection info for the first collection to match the given filter
- * @returns {void}
- */
+declare module 'mongodb-client-encryption' {
```

Comment:
These three properties are only used in the JS layer and are only added by CE and AE to communicate context to the state machine.  A future refactor could provide them as a "context" object to `StateMachine.execute` but I left as-is for now.

We use declaration merging to add the properties to the `MongoCryptContext` interface only in the driver.